release_bot/init_repo.py

```diff
@@ -94,17 +104,26 @@ def create_conf(self, silent):
         cwd = os.getcwd()
         self.release_conf['author_email'] = run_command_get_output(cwd, f'git config user.email')[1].strip()
         self.release_conf['author_name'] = run_command_get_output(cwd, f'git config user.name')[1].strip()
-
+        if self.release_conf['author_email'] == "":
+            print("WARNING: your e-mail ID from git config is not set."+
+                  "\nPlease set it using 'git config user.email \"email@example.com\"'")
+        elif not re.match(r"[^@]+@[^@]+\.[^@]+", self.release_conf["author_email"]):
+            print("WARNING: your e-mail ID from git config is not a valid e-mail address.")
+        if self.release_conf['author_name'] == "":
+            print("WARNING: your username from git config is not set." +
+                  "\nPlease set it using 'git config user.name \"John Doe\"'")
         if not silent:
             self.conf['repository_name'] = input('Please enter the repository name:')
             self.conf['repository_owner'] = input('Please enter the repository owner:')
             print("""For details on how to get github token checkout
     'https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/'""")
             self.conf['github_token'] = input('Please enter your valid github token:')
-            refresh_interval = input("""In how many seconds would you like the
+            refresh_interval = "dummy"
+            while not (refresh_interval.isdigit() or refresh_interval == ""):
+                refresh_interval = input("""In how many seconds would you like the
     bot to check for updates (Default 180):""")
             if refresh_interval:
-                self.conf['refresh_interval'] = int(refresh_interval)
+                self.conf['refresh_interval'] = refresh_interval
```

Comment:
No, you still have to convert `refresh_interval` to `int` since it's a string.

Reply:
It will be redundant in my opinion.

Since we are only writing that information in a file and not performing any numerical operations on it, we can leave it as a string.
Unless you want to change it for more uniformity in the code (since 180 - the default value, is an integer and makes it non-uniform if we store an int or string)

Reply:
But we want to write an int (not string) into the yaml file.

```
In [1]: from pathlib import Path                                                                                                                                                                                                              

In [2]: import yaml                                                                                                                                                                                                                           

In [3]: conf = {'a': '1', 'b': 2}                                                                                                                                                                                                             

In [4]: with open('testing', 'w') as yaml_file: 
   ...:     yaml.safe_dump(conf, yaml_file, default_flow_style=False) 
   ...:                                                                                                                                                                                                                                       

In [5]: print(Path('testing').read_text())                                                                                                                                                                                                    
a: '1'
b: 2
```

Reply:
but the bot will fail if it's gonna a string:
```
In [3]: time.sleep('1')                    
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-80aa3b80d707> in <module>
----> 1 time.sleep('1')
        global time.sleep = <built-in function sleep>

TypeError: an integer is required (got type str)
```

Reply:
@jpopelka , never knew that `yaml.safe_dump` will preserve the data type as well. I assumed that it everything will be stored as a string.

@TomasTomecek thanks for highlighting the error. I will fix this right away.